### PR TITLE
Refactor Dockerfile docs

### LINKED ISSUE #1668: Apply Dockerfile best practices from Snyk recommendations
I saw a nice post today from Synk about best practices for securing node apps in Docker:

https://snyk.io/blog/10-best-practices-to-containerize-nodejs-web-applications-with-docker/

Screen Shot 2021-02-07 at 9 41 11 AM

We should apply as many of these as makes sense for us to our existing Dockerfiles, and also the new ones happening in #1636 for microservices.

### PR DESCRIPTION
**Pull request by Kevan-Y**
## Issue This PR Addresses
Fixes #1668

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

- Add Docker Healthcheck
- Remove development part

## Steps to test the PR

- `cd src/docs`
- `docker build -t docs:test .`
- `docker run --rm -p 80:80 docs:test`
- Go to `localhost`
## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/docs/Dockerfile [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
-# Copy what we've installed/built from production
-COPY --chown=node:node --from=production /home/node/app/build /usr/share/nginx/html/
+# Use 1.20.2-alpine nginx image
+FROM nginx:1.20.2-alpine as deploy
+
+COPY --from=builder /app/build /usr/share/nginx/html/
+
+# Add Healthcheck
+HEALTHCHECK --interval=30s --timeout=30s --start-period=5s --retries=3 \
+  CMD curl --fail localhost:80 || exit 1
```
Comment (tcvan0707) [inline src/docs/Dockerfile:39]: One thing I want to clarify. At first, I didn't know which `localhost` should I go to (based on what you have in your **Steps to test the PR**), then I tried to specify `localhost:80` and it was successful. Is `localhost:80` same as `localhost` (when we type in the URL)?

**Review by tcvan0707 — COMMENTED** [review]
I just pulled your branch and tested locally, it was the same as how we normally run the docs `pnpm start`. Is that how we expect this PR to run?

File: src/docs/Dockerfile (hunk 1 above)
Comment (humphd) [reply src/docs/Dockerfile:39]: localhost and localhost:80 are the same, correct.

**Comment by Kevan-Y** [comment]
> I just pulled your branch and tested locally, it was the same as how we normally run the docs `pnpm start`. Is that how we expect this PR to run?

No, `pnpm start` in docs will run the development side of docusaurus,
Here we generate the static site, then we put under nginx to server our static content

**Review by menghif — APPROVED** [review]
It works

**Comment by cindyledev** [comment]
Interesting to see this change only triggers prettier and eslint ci. We want to know if the Dockerfile change builds correctly.

I think we need to update the workflow so telescope/src/docs/docs (or src/web/docusaurus/docs after we move the docs directory into src/web) triggers prettier and eslint ci.

**Comment by humphd** [comment]
Agree, https://github.com/Seneca-CDOT/telescope/blob/master/.github/workflows/continuous-integration.yml#L8 is wrong now.  @cindyledev could you file an issue?

**Comment by cindyledev** [comment]
> Agree, https://github.com/Seneca-CDOT/telescope/blob/master/.github/workflows/continuous-integration.yml#L8 is wrong now. @cindyledev could you file an issue?

No problem. I'll file it now

**Pull request merged**